### PR TITLE
Add revision log to function and API

### PR DESCRIPTION
**Pull request by FirestarJes**
Add revision log to both Orchestration version and WebAPI version of Settlement Reports

### CODE REVIEW

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportDownloadTrigger.cs [hunk 1]
```diff
@@ -43,6 +51,16 @@ public async Task SettlementReportDownload(
     {
         try
         {
+            await _revisionLogClient.LogAsync(
```
Comment (defectiveAi) [inline source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportDownloadTrigger.cs:54]: Add user and actor from IUserContext<FrontendUser> to the log.

Comment (defectiveAi) [reply source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportDownloadTrigger.cs:54]: Also, note that this violates the requirement of logging all attempts, as invalid input will not be logged.

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportListHttpTrigger.cs [hunk 2]
```diff
@@ -47,6 +53,16 @@ public async Task<HttpResponseData> ListSettlementReports(
         [DurableClient] DurableTaskClient client,
         FunctionContext executionContext)
     {
+        await _revisionLogClient.LogAsync(
```
Comment (defectiveAi) [inline source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportListHttpTrigger.cs:56]: Same two comments as before.

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportRequestTrigger.cs [hunk 3]
```diff
@@ -46,6 +52,16 @@ public async Task<HttpResponseData> RequestSettlementReport(
         [DurableClient] DurableTaskClient client,
         FunctionContext executionContext)
     {
+        await _revisionLogClient.LogAsync(
```
Comment (defectiveAi) [inline source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportRequestTrigger.cs:55]: Same two comments as before.

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportDownloadTrigger.cs (hunk 1 above)
Comment (FirestarJes) [reply source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportDownloadTrigger.cs:54]: Done

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportRequestTrigger.cs (hunk 3 above)
Comment (FirestarJes) [reply source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportRequestTrigger.cs:55]: Done

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportListHttpTrigger.cs (hunk 2 above)
Comment (FirestarJes) [reply source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/SettlementReportListHttpTrigger.cs:56]: Done

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**